model/src/test/kotlin/DependencyGraphTest.kt

```diff
@@ -0,0 +1,192 @@
+            val scopeMap = mapOf(
+                "s1" to listOf(DependencyIndex(2)),
+                "s2" to listOf(DependencyIndex(2, fragmentIndex = 1))
+            )
+
+            val graph = DependencyGraph(fragments, ids, scopeMap, MANAGER_NAME)
+            val scopes = graph.createScopes()
+
+            scopeDependencies(scopes, "s1") shouldBe "${pkgId(ids[2])}<${pkgId(ids[1])}${pkgId(ids[0])}>"
+            scopeDependencies(scopes, "s2") shouldBe "${pkgId(ids[2])}<${pkgId(ids[1])}<${pkgId(ids[3])}>" +
+                    "${pkgId(ids[0])}>"
+        }
+
+        "deal with attributes of package references" {
+            val ids = listOf(
+                id("commons-lang3", "3.10"),
+                id("commons-collections4", "4.4")
+            )
+            val issue = OrtIssue(source = "analyzer", message = "Could not analyze :-(")
+            val refLang = DependencyReference(0, linkage = PackageLinkage.PROJECT_DYNAMIC)
+            val refCol = DependencyReference(1, issues = listOf(issue), dependencies = sortedSetOf(refLang))
+            val trees = setOf(refCol)
+            val scopeMap = mapOf("s" to listOf(DependencyIndex(1)))
+
+            val graph = DependencyGraph(trees, ids, scopeMap, MANAGER_NAME)
+            val scopes = graph.createScopes()
+            val scope = scopes.first()
+
+            scope.shouldNotBeNull()
+            scope.dependencies shouldHaveSize 1
+            val pkgRefCol = scope.dependencies.first()
+            pkgRefCol.issues should containExactly(issue)
+            pkgRefCol.dependencies shouldHaveSize 1
+
+            val pkgRefLang = pkgRefCol.dependencies.first()
+            pkgRefLang.linkage shouldBe PackageLinkage.PROJECT_DYNAMIC
+        }
+
+        "correctly set the type of identifiers" {
+            val ids = listOf(
+                id("commons-lang3", "3.10"),
+                id("commons-collections4", "4.4")
+            )
+            val refLang = DependencyReference(0, dependencyType = DependencyType.UNKNOWN)
+            val refCol = DependencyReference(
+                1, dependencyType = DependencyType.MAVEN, dependencies = sortedSetOf(refLang)
+            )
+            val trees = setOf(refCol)
+            val scopeMap = mapOf("s" to listOf(DependencyIndex(1)))
+
+            val graph = DependencyGraph(trees, ids, scopeMap, MANAGER_NAME)
+            val scopes = graph.createScopes()
+
+            scopeDependencies(scopes, "s") shouldBe "${pkgId(ids[1], "Maven")}" +
+                    "<${pkgId(ids[0], "Unknown")}>"
+        }
+    }
+})
+
+/** The group of test dependencies. */
+private const val GROUP_ID = "org.apache.commons"
+
+/** The name of the dependency manager used by tests. */
+private const val MANAGER_NAME = "TestManager"
+
+/**
+ * Create an identifier string with the given [group], [artifact] ID and, [version].
+ */
+private fun id(artifact: String, version: String, group: String = GROUP_ID): String = "$group:$artifact:$version"
```

Comment:
Even if only used privately in this test, I think it's confusing to hard-code "org.apache.commons" as the group; it confused me when reading the tests as I wondered why for "junit" a group is specified but not for the others. I would prefer to not set a default argument here.